pkg/ccl/backupccl/backup_test.go

```diff
@@ -1443,6 +1443,252 @@ into_db='restoredb', %s)`, encryptionOption), backupLoc1)
+			// after the full backup ended.
+			name:      "inc-backup",
+			backupCmd: incBackupDetachedCmd,
+			preBackupWorkload: func(ctx context.Context, t *testing.T, sqlDB *sqlutils.SQLRunner) {
+				// Required to add a table for the first backup because of
+				// https://github.com/cockroachdb/cockroach/issues/115773
+				createTable(sqlDB, "t1")
+				sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE d INTO '%s'`, localFoo))
+				createTable(sqlDB, "t2")
+			},
+		},
+		{
+			// The incremental backup will _only_ back up actual data from before
+			// backupStartTime, so we expect any recorded checkpoints to be for
+			// introduced spans. NB: the tracked backup will send two export requests:
+			// one from [t0,BackupStartTime) and another for
+			// [backupStartTime,BackupEndTime). Only the first request will yield any
+			// data and thus affect checkpointing, however both requests will ping the
+			// progress channel in a non deterministic order, so it's very hard to
+			// capture the checkpoint manifest created after the non-empty response
+			// returns.
+			name:      "introduce-spans",
+			backupCmd: incBackupDetachedCmd,
+			preBackupWorkload: func(ctx context.Context, t *testing.T, sqlDB *sqlutils.SQLRunner) {
+				// Runs a workload such that the next backup of the
+				// provided table will back up all data from the table since t0.
+				tableName := "t1"
+				createTable(sqlDB, tableName)
+				sqlDB.Exec(t, `SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_publishing_descriptors'`)
+
+				// t0: Backup a table that we'll then restore with pause
+				sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE d INTO '%s'`, localFoo))
+				sqlDB.Exec(t, fmt.Sprintf(`DROP TABLE %s`, tableName))
+				var restoreJobID int
+				sqlDB.QueryRow(t, fmt.Sprintf(`RESTORE TABLE %s FROM LATEST IN '%s' with detached`, tableName, localFoo)).Scan(&restoreJobID)
+				jobutils.WaitForJobToPause(t, sqlDB, jobspb.JobID(restoreJobID))
+
+				// t1: run an inc backup while restore is paused to ensure the introduced span contains data in the subseq
+				sqlDB.Exec(t, fmt.Sprintf(`BACKUP DATABASE d INTO LATEST IN '%s'`, localFoo))
+
+				// t2: unpause the restore, allowing the next incremental backup to capture its reintroduced spans.
+				sqlDB.Exec(t, `SET CLUSTER SETTING jobs.debug.pausepoints = ''`)
+				sqlDB.Exec(t, `RESUME JOB $1`, restoreJobID)
+				jobutils.WaitForJobToSucceed(t, sqlDB, jobspb.JobID(restoreJobID))
+			},
+			checkpointsIntroducedSpans: true,
+		},
+	} {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+
+			ctx := context.Background()
+
+			// We turn on the following testing knobs during the backup whose
+			// checkpoints we want to track.k
+			exportedSpanCount := make(map[string]int)
+			var logExportSpanCount bool
+			var holdUpCheckpointing bool
+			waitForProgress := make(chan struct{})
+			waitForPause := make(chan struct{})
+
+			params := base.TestClusterArgs{}
+			var mu syncutil.Mutex
```

Comment:
Can we use our typical "mu embedded in a struct" pattern here to specify what this mutex protects (warning, code written in comment box):

```golang
	testState := struct{
		syncutil.Muxex
		logExportSpanCount bool
		holdUpCheckpointing bool
		exportedSpanCount map[string]int
	}{
		exportedSpanCount: make(map[string]int),
	}
```

Reply:
Good call. Done.